### PR TITLE
feat(m365): add CIS M365 v7.0.0 entra authentication method, PIM and access review checks

### PR DESCRIPTION
**Pull request by pedrooot**
### Context

Part of the split of #12102, which added 37 M365 checks in a single pull request. It was broken up per service so each piece can be reviewed on its own; this one covers **entra**.

The CIS Microsoft 365 Foundations Benchmark v7.0.0 framework (`cis_7.0_m365`) defines 160 controls, and 64 of its "Automated" controls had no Prowler check.

This is **part 4 of 4** of the `entra` stack (26 checks), which is split by the service getter each group of checks depends on. It is stacked on top of the previous branch, so the diff shown here is only its own.

> Merge after part 3 of the stack.

### Description

Adds **6 M365 `entra` checks** mapped to `cis_7.0_m365.json`:

| CIS control | Check |
|---|---|
| 5.2.3.1 | `entra_authentication_method_authenticator_show_context` |
| 5.2.3.7 | `entra_authentication_method_email_otp_disabled` |
| 5.3.2 | `entra_access_review_guest_users_configured` |
| 5.3.3 | `entra_access_review_privileged_roles_configured` |
| 5.3.4 | `entra_pim_global_administrator_approval_required` |
| 5.3.5 | `entra_pim_privileged_role_administrator_approval_required` |

**Data collection.** New getters (all Microsoft Graph **v1.0**) for `authenticationMethodsPolicy`, `roleManagementPolicyAssignments` (PIM) and `identityGovernance/accessReviews/definitions`.

**Not covered.** CIS 5.2.3.6 (system-preferred MFA) and 5.2.3.10 (Authenticator on companion applications) are left without checks because their settings are not exposed through the Microsoft Graph v1.0 API.

**Caveat.** PIM (5.3.4/5) and Access Reviews (5.3.2/3) require **Entra ID P2**. Parsing logic is unit-tested; the P2 paths should be confirmed against a live tenant.

Each check ships with its implementation, `metadata.json` and unit tests (PASS / FAIL / edge cases).

### Steps to review

1. **Discovery:** `uv run python prowler-cli.py m365 --list-checks | grep -E "entra_authentication_method|entra_pim_|entra_access_review"`
2. **Tests:** `uv run pytest tests/providers/m365/services/entra/`
3. **Compliance mapping:** review the 6 requirement(s) touched in `prowler/compliance/m365/cis_7.0_m365.json`.
4. **Data collection:** review the new getters and models in `entra_service.py`.
5. Spot-check the check logic against the CIS audit procedures (access reviews must be in progress and fail-closed; PIM approval must have approvers configured; guest scope is detected in both the top-level scope query and principal scopes).

### Checklist

<details>

<summary><b>Community Checklist</b></summary>

- [ ] This feature/issue is listed in the [open issues](https://github.com/prowler-cloud/prowler/issues?q=sort%3Aupdated-desc+is%3Aissue+is%3Aopen) or roadmap.prowler.com
- [ ] Is it assigned to me, if not, request it via the [open issues](https://github.com/prowler-cloud/prowler/issues?q=sort%3Aupdated-desc+is%3Aissue+is%3Aopen) or [Prowler Community Slack](https://goto.prowler.com/slack)

</details>

- [x] Review if the code is being covered by tests.
- [x] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [ ] Review if backport is needed.
- [ ] Review if is needed to change the [README.md](https://github.com/prowler-cloud/prowler/blob/master/README.md)
- [x] Ensure a changelog fragment is added under [prowler/changelog.d/](https://github.com/prowler-cloud/prowler/tree/master/prowler/changelog.d), if applicable.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

#### SDK/CLI
- Are there new checks included in this PR? **Yes** (6 M365 checks)
    - Additional read scopes required: `Policy.Read.AuthenticationMethod`, `RoleManagementPolicy.Read.Directory`, `AccessReview.Read.All`.

## Summary by CodeRabbit

* **New Features**
  * Added six Microsoft Entra security checks covering authentication methods, access reviews, and Privileged Identity Management approval requirements.
  * Added checks for Email OTP, Microsoft Authenticator context display, guest-user and privileged-role access reviews, and administrator activation approvals.
  * Added guidance for required Microsoft Graph permissions and Entra ID P2 or Governance licensing.

* **Documentation**
  * Documented the new Microsoft 365 CIS v7 checks, configuration requirements, and remediation guidance.

### CODE REVIEW

**Review by danibarranqueroo — DISMISSED** [review]

**Review by danibarranqueroo — DISMISSED** [review]

**Review by HugoPBrito — DISMISSED** [review]

**Review by HugoPBrito — DISMISSED** [review]

File: docs/user-guide/providers/microsoft365/authentication.mdx [hunk 1]
```diff
@@ -39,10 +39,12 @@ When using service principal authentication, add these **Application Permissions
 
 **Microsoft Graph API Permissions:**
 
+- `AccessReview.Read.All`: Required for `entra_access_review_guest_users_configured` and `entra_access_review_privileged_roles_configured` checks. An Entra ID P2 license is also required.
 - `AuditLog.Read.All`: Required for Entra service.
 - `Directory.Read.All`: Required for all services.
 - `OnPremDirectorySynchronization.Read.All`: Required for `entra_seamless_sso_disabled` check (hybrid deployments).
 - `Policy.Read.All`: Required for all services.
+- `RoleManagementPolicy.Read.Directory`: Required for `entra_pim_global_administrator_approval_required` and `entra_pim_privileged_role_administrator_approval_required` checks. An Entra ID P2 license is also required.
```
[the remark this answers is not in the document]
Comment (HugoPBrito) [reply docs/user-guide/providers/microsoft365/authentication.mdx]: Fixed in `0d0175a05`. The permission assignment steps now include both required Graph permissions and document the current Microsoft Entra ID P2 or Microsoft Entra ID Governance licensing options.

File: prowler/providers/m365/services/entra/entra_access_review_guest_users_configured/entra_access_review_guest_users_configured.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        top-level scope query is empty, so both are inspected.
+
+        Args:
+            definition: The access review definition to evaluate.
+
+        Returns:
+            bool: True if any of the review's scope queries reference guest users.
+        """
+        queries = [definition.scope_query] + definition.principal_scope_queries
+        return any("guest" in query.lower() for query in queries)
```
[the remark this answers is not in the document]
Comment (HugoPBrito) [reply prowler/providers/m365/services/entra/entra_access_review_guest_users_configured/entra_access_review_guest_users_configured.py]: Fixed in `0d0175a05`. Guest targeting now matches the actual case-insensitive `userType eq Guest` predicate, supports URL encoding, and rejects `ne`, direct `not(...)`, and incidental guest text with regression coverage.

File: prowler/providers/m365/services/entra/entra_access_review_guest_users_configured/entra_access_review_guest_users_configured.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        report.status_extended = (
+            "No active fail-closed access review scoped to guest users is configured."
+        )
+
+        for definition in definitions:
+            if (
+                definition.status in ACTIVE_STATUSES
+                and self._targets_guest_users(definition)
+                and self._is_fail_closed(definition)
+            ):
```
[the remark this answers is not in the document]
Comment (HugoPBrito) [reply prowler/providers/m365/services/entra/entra_access_review_guest_users_configured/entra_access_review_guest_users_configured.py:112]: Fixed in `0d0175a05`. The service now parses recurrence and primary-reviewer state, including stage reviewers with fallback-only configurations rejected. Both checks require weekly/monthly recurrence, `noEnd`, and assigned reviewers. Tests cover top-level, staged, incomplete, fallback-only, and invalid recurrence cases. The privileged-role check also now follows CIS 5.3.3 by requiring `defaultDecision == None`.

File: prowler/providers/m365/services/entra/entra_pim_global_administrator_approval_required/entra_pim_global_administrator_approval_required.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+class entra_pim_global_administrator_approval_required(Check):
+    """Check if PIM requires approval to activate the Global Administrator role.
+
+    Privileged Identity Management (PIM) should require approval to activate the
+    Global Administrator role, with at least one approver configured.
+
+    - PASS: Approval is required to activate Global Administrator and approvers exist.
+    - FAIL: Approval is not required, no approvers are configured, or PIM settings are
+      unavailable.
```
[the remark this answers is not in the document]
Comment (HugoPBrito) [reply prowler/providers/m365/services/entra/entra_pim_global_administrator_approval_required/entra_pim_global_administrator_approval_required.py]: Fixed in `0d0175a05`. The class docstring now distinguishes the FAIL conditions from unavailable settings, which produce no findings.

File: tests/providers/m365/services/entra/entra_pim_global_administrator_approval_required/entra_pim_global_administrator_approval_required_test.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        result = self._run(
+            {
+                GLOBAL_ADMINISTRATOR_ROLE_TEMPLATE_ID: PimRoleApprovalSetting(
+                    role_definition_id=GLOBAL_ADMINISTRATOR_ROLE_TEMPLATE_ID,
+                    is_approval_required=True,
+                    has_approvers=False,
+                )
+            }
+        )
+        assert result[0].status == "FAIL"
```
[the remark this answers is not in the document]
Comment (HugoPBrito) [reply tests/providers/m365/services/entra/entra_pim_global_administrator_approval_required/entra_pim_global_administrator_approval_required_test.py:54]: Fixed in `0d0175a05`. Added coverage for approval disabled with approvers present; the expected result is `FAIL`, consistent with the check logic and CIS requirement.

File: prowler/providers/m365/services/entra/entra_service.py [hunk 6]
```diff
@@ -1284,6 +1294,164 @@ async def _get_device_registration_policy(self):
...
+                        mail_notifications_enabled=bool(
+                            settings.get("mailNotificationsEnabled", False)
+                        ),
+                        reminders_enabled=bool(
+                            settings.get("reminderNotificationsEnabled", False)
+                        ),
+                        duration_in_days=settings.get("instanceDurationInDays"),
+                        recurrence_pattern_type=recurrence_pattern.get("type"),
+                        recurrence_range_type=recurrence_range.get("type"),
+                        has_primary_reviewers=has_primary_reviewers,
```
[the remark this answers is not in the document]
Comment (HugoPBrito) [reply prowler/providers/m365/services/entra/entra_service.py:1449]: Fixed in `05c89294b`. `AccessReviewDefinition` now stores `default_decision_enabled`, the service parser defaults absent values to `False`, and the guest review requires it before treating `Deny` as fail-closed. Parser and check tests cover true, false, and absent settings.

**Review by HugoPBrito — APPROVED** [review]
